internal/jujuapi/role.go

```diff
@@ -0,0 +1,124 @@
+func (r *controllerRoot) AddRole(ctx context.Context, req apiparams.AddRoleRequest) (apiparams.AddRoleResponse, error) {
+	const op = errors.Op("jujuapi.AddRole")
+	resp := apiparams.AddRoleResponse{}
+
+	if !jimmnames.IsValidRoleName(req.Name) {
+		return resp, errors.E(op, errors.CodeBadRequest, "invalid role name")
+	}
+
+	roleEntry, err := r.jimm.GetRoleManager().AddRole(ctx, r.user, req.Name)
+	if err != nil {
+		zapctx.Error(ctx, "failed to add role", zaputil.Error(err))
+		return resp, errors.E(op, err)
+	}
+	resp = apiparams.AddRoleResponse{Role: apiparams.Role{
+		Name:      roleEntry.Name,
+		UUID:      roleEntry.UUID,
+		CreatedAt: roleEntry.CreatedAt.Format(time.RFC3339),
+		UpdatedAt: roleEntry.UpdatedAt.Format(time.RFC3339),
+	}}
+
+	return resp, nil
+}
+
+// GetRole returns role information based on a UUID or name.
+func (r *controllerRoot) GetRole(ctx context.Context, req apiparams.GetRoleRequest) (apiparams.Role, error) {
+	const op = errors.Op("jujuapi.GetRole")
+
+	var roleEntry *dbmodel.RoleEntry
+	var err error
+	switch {
+	case req.UUID != "" && req.Name != "":
+		return apiparams.Role{}, errors.E(op, errors.CodeBadRequest, "only one of UUID or Name should be provided")
+	case req.Name != "" && !jimmnames.IsValidRoleName(req.Name):
+		return apiparams.Role{}, errors.E(op, errors.CodeBadRequest, "invalid role name")
+	case req.UUID != "":
+		roleEntry, err = r.jimm.GetRoleManager().GetRoleByUUID(ctx, r.user, req.UUID)
+	case req.Name != "":
+		roleEntry, err = r.jimm.GetRoleManager().GetRoleByName(ctx, r.user, req.Name)
+	default:
+		return apiparams.Role{}, errors.E(op, errors.CodeBadRequest, "no UUID or Name provided")
+	}
+	if err != nil {
+		zapctx.Error(ctx, "failed to get role", zaputil.Error(err))
+		return apiparams.Role{}, errors.E(op, err)
+	}
+
+	return apiparams.Role{
+		UUID:      roleEntry.UUID,
+		Name:      roleEntry.Name,
+		CreatedAt: roleEntry.CreatedAt.Format(time.RFC3339),
+		UpdatedAt: roleEntry.UpdatedAt.Format(time.RFC3339),
+	}, nil
+}
+
+// RenameRole renames a role within JIMMs DB for reference by OpenFGA.
+func (r *controllerRoot) RenameRole(ctx context.Context, req apiparams.RenameRoleRequest) error {
+	const op = errors.Op("jujuapi.RenameRole")
+
+	if !jimmnames.IsValidRoleName(req.NewName) {
+		return errors.E(op, errors.CodeBadRequest, "invalid role name")
+	}
+
+	if err := r.jimm.GetRoleManager().RenameRole(ctx, r.user, req.Name, req.NewName); err != nil {
+		zapctx.Error(ctx, "failed to rename role", zaputil.Error(err))
+		return errors.E(op, err)
+	}
+	return nil
+}
+
+// RemoveRole removes a role within JIMMs DB for reference by OpenFGA.
+func (r *controllerRoot) RemoveRole(ctx context.Context, req apiparams.RemoveRoleRequest) error {
+	const op = errors.Op("jujuapi.RemoveRole")
+
+	if !jimmnames.IsValidRoleName(req.Name) {
+		return errors.E(op, errors.CodeBadRequest, "invalid role name")
+	}
+
+	if err := r.jimm.GetRoleManager().RemoveRole(ctx, r.user, req.Name); err != nil {
```

Comment:
Also if we do go with Getter, in go you don't prefix getters with Get - think essential go or something else states this